File: LeetCode/2520_Count_the_Digits_That_Divide_a_Number/solution.js
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function(num) {
    let count=0;
    let n=num;
    while(n>0){
        let a = n%10;
        n=n-a;
        n/=10;
        if(num%a==0) count++;
    }
    return count;
};

File: LeetCode/2520_Count_the_Digits_That_Divide_a_Number/solution.cpp
class Solution {
public:
    int countDigits(int num) {
        int count = 0;
        int n = num;
        while (n > 0) {
            int a = n % 10;
            n = n - a;
            n /= 10;
            if (num % a == 0)
                count++;
        }
        return count;
    }
};